### ISSUE TITLE
Ability to Duplicate/Resubmit instance

### ISSUE THREAD

**Issue by Asharif88** [enketo/enketo-express#93]
Hi,
We are working with Enketo and Kobotoolbox as part of a bigger project, one feature we're trying to achieve is the ability to maybe duplicate an instance, allow the user to edit it then submit it as a new one, we already tried with the post survey API endpoint, while it works fine with a few fields, our survey is around 500-600 questions so loading the values through the request URI quickly becomes too big for the browsers.
Sorry if this is not where it should be posted. Appreciate responses.

**Comment by MartijnR** [comment]
> becomes too big for the browsers.

How does this problem manifest itself?

**Comment by MartijnR** [comment]
Thanks. I think this (relatively new) setting may help with that: https://github.com/enketo/enketo-express/tree/master/config#payload-limit

**Comment by Asharif88** [comment]
Thanks, yes this helps to some extent, but i'm not sure if it's sustainable (eventually we want the URL to be opened on a tablet).
Is there a way to have the edit instance add a new row? if not is it planned or in the roadmap? i haven't seen it mentioned in the forums so far.

**Comment by MartijnR** [comment]
> Is there a way to have the edit instance add a new row? 

A new repeat instance? If so, that would be for all repeats in the form (incl nested, but excluding any repeats that have a repeat-coutn)? That would seem best implemented as a custom feature in a fork (unless I'm missing the general usefulness of this). 

I cannot immediately think of a very elegant way of doing this, but a hack would be to add a line in Enketo Express that clicks all + buttons upon load.

**Comment by Asharif88** [comment]
Hi, i might have used mixed terminology, sorry for that, a little background and more details:
- we have a big questionnaire (around 600 questions), that gets filled in the field.
- It needs to be refilled every few months.
- The idea is that the teams can save time by opening the last submission for this location they're surveying and just modifying what has changed since they we last there.
- In order to be able to track the history of the location we need the edited submission as a new one (so we can keep the old one).
So basically what we're looking for is something almost exact to what happens when you click on "Edit" in the form data row, but when you finish editing and click "Save" it goes as a new submission rather that overwriting the old one.
Hope it's clear and many thanks for help.

**Comment by MartijnR** [comment]
Ah, thanks, I got it. There are some ongoing discussions about adding this formally (case management, longitudinal data) within the ODK ecosystem (incl KoBo), but it's all quite preliminary. 

Technically, you could have a look at /meta/instanceID and /meta/deprecatedID in edited submissions. The original instanceID becomes a deprecatedID and a new instanceID is generated. This is how KoBo keeps track of what is an edit and what is a new submission. Tweaking this mechanism could allow you to fool KoBo into thinking the edited submission is a new submission. You'd have to figure out how to match entities in your analysis though (in your case locations). The latter is the main thing we're trying to figure out within ODK before adding this feature to the specs.

**Comment by MartijnR** [comment]
This will be replaced with full-fledged case management support after this has been agreed up on within ODK.
